.gitlab/build.yml

```diff
@@ -0,0 +1,37 @@
+build:
+  timeout: 2 hours
+  image:
+    name: kiltprotocol/kilt-ci:2.7.31
+    entrypoint: [""]
+  stage: build
+  only:
+    refs:
+      - develop
+      - master
+      - /^[0-9]+(?:\.[0-9]+){2}(?:-(rc)*([0-9])+)?$/
+      - /^[dev-](\w*-)*[0-9]?$/
+  variables:
+    DOCKER_HUB_PARACHAIN: "kiltprotocol/kilt-node"
+    DOCKER_HUB_STANDALONE: "kiltprotocol/standalone-node"
+    DOCKER_HUB_DIP_PROVIDER_TEMPLATE: "kiltprotocol/dip-provider-node-template"
+    DOCKER_HUB_DIP_CONSUMER_TEMPLATE: "kiltprotocol/dip-consumer-node-template"
+  before_script:
+    - aws --version
+    - docker --version
+  script:
+    - echo -n $CI_REGISTRY_TOKEN | docker login -u "$CI_REGISTRY_USER" --password-stdin $CI_REGISTRY
+    - aws ecr get-login-password | docker login --username AWS --password-stdin $AWS_REGISTRY
+    - ./.maintain/docker-auth-config.sh
+    - ./.maintain/build-image.sh build
+    - "if [[ ! -z ${CI_COMMIT_BRANCH} ]]; then ./.maintain/push-image.sh build ${CI_COMMIT_SHORT_SHA}; fi"
+    - "if [[ ! -z ${CI_COMMIT_BRANCH} ]]; then ./.maintain/push-image.sh build latest-${CI_COMMIT_BRANCH}; fi"
+    - "if [[ ! -z ${CI_COMMIT_TAG} ]]; then ./.maintain/push-image.sh build ${CI_COMMIT_TAG}; fi"
+    - "if [[ ! -z ${CI_COMMIT_TAG} && ! -z ${CI_COMMIT_TAG##*-rc*} && ! -z ${CI_COMMIT_TAG##*dev-*} ]]; then ./.maintain/push-image.sh build latest; fi"
+  cache:
```

Comment:
We use `docker build` here to build the image. Does the cache still work? I don't think it does anything if we cache the local Rust stuff, does it?

Reply:
It could have been possible to pass `docker build --build-arg CARGO_CACHE_PATH=$CARGO_CACHE_PATH` yet it seems overkill for a job that is rarely triggered. Removing cache here